### ISSUE TITLE
uninstall and reinstall flow with and without kfctl

### ISSUE THREAD

**Issue by ryandawsonuk** [kserve/kserve#481]
I tried to uninstall by deleting the statefulset, mutating webhook, validating webhook and the CRD. I then reinstalled from kfctl. But then the mutator starts failing for every pod deployed and the logs show:
```
{"level":"info","ts":1571911982.801687,"logger":"kubebuilder.controller","msg":"Starting workers","controller":"kfserving-controller","worker count":1}
2019/10/24 10:13:29 http: TLS handshake error from 127.0.0.1:39972: EOF
2019/10/24 10:13:30 http: TLS handshake error from 127.0.0.1:39978: EOF
2019/10/24 10:13:30 http: TLS handshake error from 127.0.0.1:39984: EOF
2019/10/24 10:13:30 http: TLS handshake error from 127.0.0.1:39996: EOF
2019/10/24 10:13:30 http: TLS handshake error from 127.0.0.1:40002: EOF
2019/10/24 10:13:30 http: TLS handshake error from 127.0.0.1:40012: EOF
2019/10/24 10:13:31 http: TLS handshake error from 127.0.0.1:40018: EOF
2019/10/24 10:13:31 http: TLS handshake error from 127.0.0.1:40022: EOF
2019/10/24 10:13:31 http: TLS handshake error from 127.0.0.1:40038: EOF
2019/10/24 10:13:33 http: TLS handshake error from 127.0.0.1:40070: EOF
2019/10/24 10:13:36 http: TLS handshake error from 127.0.0.1:40142: EOF
2019/10/24 10:13:41 http: TLS handshake error from 127.0.0.1:40280: EOF
2019/10/24 10:13:51 http: TLS handshake error from 127.0.0.1:40550: EOF
2019/10/24 10:14:12 http: TLS handshake error from 127.0.0.1:41066: EOF
2019/10/24 10:14:53 http: TLS handshake error from 127.0.0.1:42076: EOF
2019/10/24 10:15:29 http: TLS handshake error from 127.0.0.1:42986: EOF
2019/10/24 10:16:15 http: TLS handshake error from 127.0.0.1:44116: EOF
```
I also tried a `kfctl delete` and that uninstalled kfserving (along with other kubeflow components). Then I did a `kfctl apply` and then it couldn't even create the controller manager pod (or any other pods). A describe on the statefulset reveals `failed calling webhook "inferenceservice.kfserving-webhook-server.pod-mutator": Post https://kfserving-webhook-server-service.kubeflow.svc:443/mutate-pods?timeout=30s: service "kfserving-webhook-server-service" not found`

**Comment by yuzisun** [comment]
you should run `make undeploy` which deletes the mutatingwebhookconfigration, otherwise apiserver will still try to call the webook

**Comment by ryandawsonuk** [comment]
I deleted the hooks with `kubectl` and with `kfctl`. I guess like https://github.com/kubeflow/kfserving/issues/480 this problem is specific to a `kfctl` install.

**Comment by yuzisun** [comment]
I think kfctl delete needs to handle mutatingwebhookconfiguration deletion.

**Comment by jeffgriffith** [comment]
@jlewi I think i encountered this error and several more in dealing with a cluster I needed to uninstall/reinstall (TLS handshake, "cannot start configuration manager", kfserving-controller-manager would not start). These were very hard for me to debug because there wasn't a lot of help in the events to point me in the right direction. I did finally make things work however. Here was my summary from a discussion with @yuzisun:

**KF 1.0.1 on K8S 1.15.7**

- I failed my first install because the cluster had its own cert-manager pre-installed which was pre-0.11 and it causes seldon-core-controller to fail hard.
- Uninstalled the best I could. There doesn't seem to be a clear uninstall strategy. I did a delete with kfctl but it seems to leave kubeflow CRDs around. I deleted istio-system and knative-serving resources and namespaces manually since i wasn't sure how to go about it properly.
- Somewhere in those leftovers cause the 'Failed to start configuration manager' errors mentioned in k8s 1.14 install previously which prevents knative-serving from coming up.
- Finally ironed out all that manual cleanup (deleting all the various CRDs manually).
- Still unable to start kfserving-controller-manager but with no details to help in terms of events. Since i never saw the pod for it, i felt like it had to be webhook related. (only events were "failed to create in a timely manner")
- Found issue 580 which is closed because it says it was fixed in 0.2.2 but I tried it anyway. Finally tried another install with the deletions i found in that ticket and it worked.
